models/Keyword.yaml

```diff
@@ -1 +1,2 @@
-# pk: String
+type: String
```

Comment:
firstly, please document the types here: statement, expression, type - I can guess what the first two mean, but a "type keyword" doesnt sound familiar to me from language theory.  I need to look at the data to see that you are using it for datatype names.

Doing it this way creates a burden on people creating new language definitions, because they will need to update the `_all.yaml` list when ever the language adds new keywords, which nearly every language does.

And it will mean there are constant merge conflicts on that file.

But more importantly, what has been gained?  Very little im afraid.

The reason to model something is because it will give us some way to understand the content of the input text better, or (and this is more important at the moment) understand the language better.

So rather than use an enum like enhancement to group keywords, create a new language property.
The reason that delimiters use an enum is that all delimiters are used to break up the text - they are used for tokenisation, and as such it is desirable that they all use the same model.

Add `datatypes`.

And dont try to describe all datatypes of all languages -- leave them as a data set which is automatically managed by statik.

You can try to create a very basic model for a datatype, and build a few instances, but dont try to do them all, and dont try to precisely describe a datatype.  Think of *some* common attributes of a datatype - design only the Datatype attributes ones which you fully understand.  And dont worry if the model doesnt completely describe the datatypes.  e.g. my delimiter for yaml and python is almost unusable.  That is ok, you will get more tasks which can be used to improve the model.

In effect these would have two ways of being expressed in the yaml.

If the `datatypes` list doesnt have a data file, the `pk` is the keyword, and no extra information is known about it.

If the `datatypes` list entry has a data file, the information about the datatype exists in the data file, and one of the most important fields in that model will be 'keyword', so that it is always possible to know the keyword of the datatype.

Then the problem is that `int` , `str` and `string` will appear in many languages.  clashes like this are not important when no extra information is provided about the datatype.

However when there is a data file for the datatype, the `pk, and filename, need to different.

i.e. `int_python.yaml` and `int_c.yaml` would have quite different data in them.

Which means that if there are two languages with an identical datatype which each use a different keyword for that same datatype, there would be two datatype entries.  That isnt ideal, but because the files are yaml it is easy to detect these and thus know that the two languages have the same datatype using a different keyword.  e.g. `str_python3.yaml` and `unicode_python2.yaml`.

Any entry in `datatypes` which doesnt match a data file will result in a warning in the build log.  This is great.
Only update a few languages.  Then other participants can do the remaining languages, using warnings in the build log as a guide.

Then leave the other types of keywords in `keywords`.  Someone else can tackle those as their 'create new property' task.

Reply:
Yeah, just classifying the keyword doesn't give us any information about the language itself. It was my misunderstanding with the task, sorry.

Anyway, so you mean that leave classifying all the keywords, but extract the only the datatypes and make it into a new attribute right?

I really appreciate you sincere review. thanks again.